posts/postRouter.js

```diff
@@ -1,27 +1,91 @@
-const express = 'express';
+const express = require("express");
 
-const router = express.Router();
+const postDb = require("./postDb");
 
-router.get('/', (req, res) => {
+const router = express.Router();
 
+router.get("/", async (req, res) => {
+  try {
+    const allPosts = await postDb.get();
+    if (allPosts.length > 0) {
+      res.status(200).json(allPosts);
+    } else {
+      res.status(500).json({ message: "Failed to get array of posts" });
+    }
+  } catch (error) {
+    res.status(500).json(error);
+  }
 });
 
-router.get('/:id', (req, res) => {
-
+router.get("/:id", validatePostId, (req, res) => {
+  if (req.post) {
+    res.status(200).json(req.post);
+  } else {
+    res
+      .status(500)
+      .json({ message: "Should not be getting here in get post by id" });
+  }
 });
 
-router.delete('/:id', (req, res) => {
-
+router.delete("/:id", validatePostId, async (req, res) => {
+  try {
+    const successDelete = await postDb.remove(req.post.id);
+    if (successDelete) {
+      res.status(204).end();
+    } else {
+      res
+        .status(500)
+        .json({ message: "Something went wrong when deleting user" });
+    }
+  } catch (error) {
+    res.status(500).json(error);
+  }
 });
 
-router.put('/:id', (req, res) => {
-
+router.put("/:id", validatePostId, validatePost, async (req, res) => {
+  try {
+    const addPost = await postDb.update(req.post.id, req.newPost);
```

Comment:
addPost is a little confusing here, I understand you probably copied this but on top of that tense would be helpful since this sounds like a method to be used to add a post `addedPost` or better yet, `updatedPost` in this case would help someone reading your code, which your boss will do, a lot.